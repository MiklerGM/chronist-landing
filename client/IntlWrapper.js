import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';

import localeDataRU from './locales/ru.json';
// import localeDataEN from './locales/en.json';

class IntlWrapper extends React.Component {
  render() {
    return (
      <IntlProvider
        locale={this.props.locale}
        key={this.props.locale}
        messages={this.props.messages}
      >
        {this.props.children}
      </IntlProvider>
    );
  }
}

IntlWrapper.propTypes = {
  locale: PropTypes.string,
  messages: PropTypes.object
};

IntlWrapper.defaultProps = {
  locale: 'ru',
  messages: localeDataRU
};

export default IntlWrapper;
