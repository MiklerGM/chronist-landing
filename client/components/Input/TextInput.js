import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

class TextInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      valid: true
    };
  }

  handleInvalid(e) {
    e.preventDefault();
    this.setState({ valid: false });
    this.forceUpdate();
  }

  render() {
    return (
      <div className='input--cell'>
        <FormattedMessage id={this.props.placeholder}>
          { placeholder =>
            <input
              type='text'
              value={this.props.value}
              placeholder={placeholder}
              className={(this.state.valid) ? '' : 'invalid'}
              maxLength='80'
              // pattern='^[a-zA-Zа-яА-я0-9._%+-]'
              onInvalid={e => this.handleInvalid(e)}
              onChange={(e) => {
                this.setState({ valid: true });
                this.props.cb({ [this.props.name]: e.target.value });
              }}
              required
            />
          }
        </FormattedMessage>
        <span
          style={{ gridRow: '2 / 3' }}
          className={(this.state.valid)
            ? 'invalid-message__hidden' : 'invalid-message'}
        >
          <FormattedMessage id='input.textarea.invalid' />
        </span>
      </div>
    );
  }
}

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cb: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default TextInput;
