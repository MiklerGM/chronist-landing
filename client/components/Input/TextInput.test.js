import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import IntlWrapper from '../../IntlWrapper';
import TextInput from './TextInput';

const TestComponent = () => (
  <IntlWrapper>
    <TextInput
      value=''
      name='email'
      cb={e => console.log('dumb func', e)}
      placeholder='contact.placeholder.email'
    />
  </IntlWrapper>
);

// it must render
it('TextInput render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

it('TextInput render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('TextInput Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
