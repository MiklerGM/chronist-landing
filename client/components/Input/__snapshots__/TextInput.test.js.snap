// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Snapshot 1`] = `
<div
  className="input--cell"
>
  <input
    className=""
    maxLength="80"
    onChange={[Function]}
    onInvalid={[Function]}
    placeholder="Электронная почта"
    required={true}
    type="text"
    value=""
  />
  <span
    className="invalid-message__hidden"
    style={
      Object {
        "gridRow": "2 / 3",
      }
    }
  >
    <span>
      Необходимо ввести текст
    </span>
  </span>
</div>
`;
